packages/react/src/components/Notification/next/Notification.js

```diff
@@ -0,0 +1,536 @@
+/**
+ * Copyright IBM Corp. 2016, 2018
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import PropTypes from 'prop-types';
+import React, { useEffect, useRef, useState } from 'react';
+import cx from 'classnames';
+import { settings } from 'carbon-components';
+import {
+  Close20,
+  ErrorFilled20,
+  CheckmarkFilled20,
+  WarningFilled20,
+  WarningAltFilled20,
+  InformationFilled20,
+  InformationSquareFilled20,
+} from '@carbon/icons-react';
+
+import Button from '../../Button';
+import useIsomorphicEffect from '../../../internal/useIsomorphicEffect';
+import { useNoInteractiveChildren } from '../../../internal/useNoInteractiveChildren';
+import { keys, matches } from '../../../internal/keyboard';
+
+const { prefix } = settings;
+
+/**
+ * Conditionally call a callback when the escape key is pressed
+ * @param {func} callback - function to be called
+ * @param {bool} override - escape hatch to conditionally call the callback
+ */
+function useEscapeToClose(callback, override = true) {
+  const handleKeyDown = (event) => {
+    if (matches(event, [keys.Escape]) && override) {
+      callback(event);
+    }
+  };
+
+  useIsomorphicEffect(() => {
+    document.addEventListener('keydown', handleKeyDown, false);
+    return () => document.removeEventListener('keydown', handleKeyDown, false);
+  });
+}
+
+export function NotificationActionButton({
+  children,
+  className: customClassName,
+  onClick,
+  notificationType,
+  ...rest
+}) {
+  const className = cx(
+    customClassName,
+    `${prefix}--${notificationType}-notification__action-button`
+  );
+
+  return (
+    <Button
+      className={className}
+      kind="ghost"
+      onClick={onClick}
+      size="small"
+      {...rest}>
+      {children}
+    </Button>
+  );
+}
+
+NotificationActionButton.propTypes = {
+  /**
+   * Specify the content of the notification action button.
+   */
+  children: PropTypes.node,
+
+  /**
+   * Specify an optional className to be applied to the notification action button
+   */
+  className: PropTypes.string,
+
+  /**
+   * Specify the notification type
+   */
+  notificationType: PropTypes.oneOf(['inline']),
```

Comment:
Is this prop needed if there is only one type of notification for this component? I'm a little confused about what it does for this if there aren't other options.

Reply:
It previously had another option of `persistent` for the PersistentNotification but since we're waiting on that I removed it and forgot to delete the prop altogether. Thanks for the catch!